src/fondant/pipeline/pipeline.py

```diff
@@ -145,13 +146,18 @@ def __init__(
         cache: t.Optional[bool] = True,
         resources: t.Optional[Resources] = None,
         component_dir: t.Optional[Path] = None,
+        dataset_fields: t.Optional[t.Mapping[str, Field]] = None,
```

Comment:
I'm wondering if we shouldn't split all this `consumes` / `produces` logic into a separate class, eg. a `ComponentOpFactory`. I think it could help with maintainability and testability to keep the `ComponentOp` limited to a representation of the operation, close to a dataclass.